crates/notion-core/src/tool/npm.rs

```diff
@@ -0,0 +1,69 @@
+use std::env::{args_os, ArgsOs};
+use std::ffi::{OsStr, OsString};
+use std::io;
+use std::process::{Command, ExitStatus};
+
+use super::{
+    arg0, command_for, display_error, intercept_global_installs, NoGlobalInstallError,
+    NoSuchToolError, Tool,
+};
+use session::{ActivityKind, Session};
+
+use notion_fail::Fallible;
+
+/// Represents a `npm` executable.
+pub struct Npm(Command);
+
+impl Tool for Npm {
+    fn new(session: &mut Session) -> Fallible<Self> {
+        session.add_event_start(ActivityKind::Npm);
+
+        let mut args = args_os();
+        let exe = arg0(&mut args)?;
+
+        if intercept_global_installs() {
+            let mut search_args = args_os().skip(1);
+            let command = search_args.next();
+
+            // npm global installs will be of the form `npm i` or `npm install`
+            // with `-g` or `--global` somewhere in the arguments
+            if (command == Some(OsString::from("i")) || command == Some(OsString::from("install")))
```

Comment:
Ooh, and the plot thickens: `npm` also accepts `isntall` as an [undocumented alias](https://github.com/npm/npm/issues/2933).

You can find the complete list of `install` aliases [in the source](https://github.com/npm/cli/blob/36682d4482cddee0acc55e8d75b3bee6e78fff37/lib/config/cmd-list.js). I see the following aliases:

- `i`
- `isntall`
- `add`

I know this is icky, but it's what we're signing up for by intercepting global installations. We'll need to track the evolution of `npm` and `yarn` and intercept any new aliases as they come into existence.

(We could get _super_ tricky and cross-reference the alias list for each version of `npm`. But since all we're doing is failing fast, I think it's fine to just intercept the full set of all aliases regardless of what version of `npm` is being used.)